app/src/test/java/org/mozilla/fenix/library/downloads/DownloadControllerTest.kt

```diff
@@ -33,10 +35,23 @@ class DownloadControllerTest {
     private val scope = TestCoroutineScope()
     private val store: DownloadFragmentStore = mockk(relaxed = true)
     private val state: DownloadFragmentState = mockk(relaxed = true)
-    private val openToFileManager: (DownloadItem, BrowsingMode?) -> Unit = mockk(relaxed = true)
-    private val displayDeleteAll: () -> Unit = mockk(relaxed = true)
-    private val invalidateOptionsMenu: () -> Unit = mockk(relaxed = true)
-    private val deleteDownloadItems: (Set<DownloadItem>) -> Unit = mockk(relaxed = true)
+
+    private val openToFileManager: (DownloadItem, BrowsingMode?) -> Unit = { item, mode ->
+        openToFileManagerCapturedItem = item
+        openToFileManagerCapturedMode = mode
+    }
+    private var openToFileManagerCapturedItem: DownloadItem? = null
+    private var openToFileManagerCapturedMode: BrowsingMode? = null
```

Comment:
Hm I guess these are now shared between tests and we're not setting them to null in `@Before`, which could be another source of intermittents

Reply:
Good catch.

I don't know if this is standard behavior but I added `println(openToFileManagerCapturedItem)` as the first line to each test and they all print `null`. Similarly, I added `init { println("lol") }` to the class definition and it gets called before each test. It appears the class gets re-initialized before every test so that these are actually nulled before every test. I didn't think junit worked like this (from many years ago) but I've noticed the behavior in fenix before. Maybe it's a robolectric thing? Or a more recent android thing?

I'll make the change but it's curious that it works, huh?

Reply:
@mcomella Yes, this works because JUnit by default creates a new instance of the test class before executing a test method. So, this is actually fine, by default, and be configured using `TestInstance`: https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/TestInstance.html

I also have a memory of having issues with this, but in your case it's actually fine!

Reply:
It says `TestInstance` defaults to `PER_METHOD` whose docs say:
> When using this mode, a new test instance will be created for each test method or test factory method.
>
>This mode is analogous to the behavior found in JUnit versions 1 through 4.

So I guess I won't change it then given it's less redundant? I wonder why we both have memories of this causing problems though…